hw3/kottans/lib/kottans/router.rb

```diff
@@ -0,0 +1,58 @@
+class Router
+
+  def call(env)
+    tmp_hsh = @routes[env['REQUEST_METHOD']].select do |k, v|
+      @patterns[k] === env['REQUEST_PATH']
+    end
+    tmp_hsh.values.first.call(env)
+  rescue
```

Comment:
It's better to `rescue` from some specific exception
http://daniel.fone.net.nz/blog/2013/05/28/why-you-should-never-rescue-exception-in-ruby/